### PR TITLE
Add support for secret exists

### PR DESCRIPTION
**Pull request by ygalblum**
Add the command along with the abi and tunnel support Add e2e tests
Add man page

#### Does this PR introduce a user-facing change?
Yes

```release-note
New command - podman secret exist <name>
```

### CODE REVIEW

File: pkg/api/server/register_secrets.go [hunk 1]
```diff
@@ -89,6 +89,25 @@ func (s *APIServer) registerSecretHandlers(r *mux.Router) error {
...
+	// parameters:
+	//  - in: path
+	//    name: name
+	//    type: string
+	//    required: true
+	//    description: the name or ID of the secret
+	// produces:
+	// - application/json
+	// responses:
+	//   '200':
```
Comment (vrothberg) [inline pkg/api/server/register_secrets.go:106]: I think that a `204` for the positive and a `404` for the negative case would be more RESTful.  That's what the other `/exists` endpoints do as well AFAICS.

**Review by vrothberg — COMMENTED** [review]
Thanks, @ygalblum!

Other than the nit on the http codes, LGTM

File: pkg/api/server/register_secrets.go (hunk 1 above)
Comment (ygalblum) [reply pkg/api/server/register_secrets.go]: Not a samll nit :)
Fixed and adjusted both the server and client sides

**Review by vrothberg — APPROVED** [review]
LGTM

@containers/podman-maintainers PTAL

**Review by Luap99 — COMMENTED** [review]
Can you add a test in `test/apiv2` and actually check that the HTTP code is 204 and 404 just the prevent someone from changing it by accident.

File: cmd/podman/secrets/exists.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+var (
+	existsCmd = &cobra.Command{
+		Use:               "exists SECRET",
+		Short:             "Check if a secret exists in local storage",
+		Long:              `If the named secret exists in local storage, podman secret exists exits with 0, otherwise the exit code will be 1.`,
+		Args:              cobra.ExactArgs(1),
+		RunE:              exists,
+		ValidArgsFunction: common.AutocompleteSecrets,
+		Example: `podman secret exists ID
+  podman secret exists SECRET && podman secret create SECRET <secret source>`,
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply cmd/podman/secrets/exists.go]: You are correct. Thanks for the catch

**Comment by ygalblum** [comment]
> Can you add a test in `test/apiv2` and actually check that the HTTP code is 204 and 404 just the prevent someone from changing it by accident.

Done

**Review by Luap99 — COMMENTED** [review]
/lgtm
/hold

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Pull request merged**